presto-spi/src/main/java/io/prestosql/spi/connector/TableScanRedirectApplicationResult.java

```diff
@@ -26,12 +26,27 @@
     private final Map<ColumnHandle, String> destinationColumns;
     // filter that needs to be applied on top of table scan
     private final TupleDomain<String> filter;
+    // should coercions be used when source and redirected column types don't match
+    private boolean allowCoercions;
```

Comment:
When and why would this ever be false? This doesn't seem necessary. If the connector doesn't want to allow coercions, it should just not perform the redirection.

Reply:
The can do this on its own. It needs to somehow be aware of the target table anyway, so it can do this check by itself.

Reply:
>  It needs to somehow be aware of the target table anyway

It might not be aware of target table types since it can be in different catalog and could be created externally.

Reply:
Then how is it going to know to redirect to those tables, or that they are the right table to redirect to? Table redirections don't seem very useful without that level of understanding, so it seems reasonable to me to expect that from connectors.

Reply:
removed toggle